### DISCUSSION TITLE
OpenAPI specs and Swagger for all Kong architecture

### DISCUSSION THREAD

**Discussion by ilbertt** [supabase/supabase#9211] [Questions] [answered]
Following [Generating Types](https://supabase.com/docs/guides/api/generating-types#generating-types-from-openapi-specification) instructions, I can retrieve **OpenAPI specs** for `/rest/v1` (_PostgREST_) endpoint only.
Instead, it would be nice to have a full specification also for all other endpoints of [Supabase Architecture](https://supabase.com/docs/architecture), because sometimes it's not possible to use Supabase client libraries (in [k6](https://k6.io/) load tests for example, where **k6** is a powerful load test tool [suggested also by Supabase docs](https://supabase.com/docs/going-into-prod#performance) themselves) and the only way to communicate with Supa infrastructure is through raw HTTP.

Is there a way to retrieve a full OpenAPI specification for all Supabase Kong endpoints?

From docs, I found there is already an [explanation](https://supabase.com/docs/reference/storage/usage) of `/storage` endpoint, together with an [OpenAPI spec](https://supabase.github.io/storage-api/). What about other endpoints (`/auth`, `/realtime`, ...)?

**Comment by ilbertt** [comment, marked as answer]
Actually I found the `/auth` specification at https://github.com/supabase/gotrue#endpoints, but it would be great to also have a Swagger page like for `/storage`.
Also, I wasn't able to connect to `/auth`, `/storage`, ... endpoints until I figured out they need `/v1` path, which wasn't explained anywhere. I only found it in these [supabase-js SupabaseClient.ts](https://github.com/supabase/supabase-js/blob/master/src/SupabaseClient.ts#L72-L75) file lines.

**Reply by ilbertt** [reply]
I also found Swagger spec for `/auth` endpoint at https://github.com/supabase/gotrue/blob/master/swagger.json, and parsed it with https://editor-next.swagger.io

It would be great to have them all integrated in a supabase page!

**Reply by steve-chavez** [reply]
> It would be great to have them all integrated in a supabase page!

@Luca8991 Fuly agree. I've forwarded this to our docs team. Thanks!
